src/GraphQL.MicrosoftDI/ScopedResolveConnectionContextAdapter.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using GraphQL.Builders;
+using GraphQL.Execution;
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+
+namespace GraphQL.MicrosoftDI
+{
+    internal sealed class ScopedResolveConnectionContextAdapter<TSource> : IResolveConnectionContext<TSource>
+    {
+        private readonly IResolveConnectionContext<TSource> _baseContext;
+
+        public ScopedResolveConnectionContextAdapter(IResolveConnectionContext<TSource> baseContext, IServiceProvider serviceProvider)
```

Comment:
Why `IResolveConnectionContext<TSource>` ? I see non-generic `IResolveFieldContext` in the corresponding `ScopedResolveFieldContextAdapter<TSource>`.

Reply:
This class is `internal`, and can only ever be used in a typed context, so I simplified the code.

`ScopedResolveFieldContextAdapter` (also `internal`) is used by `ScopedFieldResolver` & `ScopedAsyncFieldResolver` and requires an untyped `IResolveFieldContext` as its input.

Reply:
I promised to review this PR again. To do this, I once again looked at the code of existing builders. I don't use connection builder at my work so I can misunder about the needs. I believe that all builders must have the most similar design, including for generic constraints (or not having them). I made some changes and show them in another PR. Most likely, some of them will need to roll back.

Reply:
>  and can only ever be used in a typed context, so I simplified the code.

OK. I agree.

>  and requires an untyped IResolveFieldContext as its input.

See #2427 . I would like to get such a solution when the design all builders was the same.